src/containers/TransferCurrency/TransactionPage.tsx

```diff
@@ -0,0 +1,258 @@
+import { goBack, push } from 'connected-react-router'
+import * as React from 'react'
+import { connect } from 'react-redux'
+
+import {
+  addContact,
+  insufficientFunds,
+  showNotification,
+  updateContactForm,
+  updateRemainingBalance,
+  updateTransferForm,
+} from '@actions'
+
+import * as kagLogo from '@icons/kag-icon.svg'
+import * as kauLogo from '@icons/kau-icon.svg'
+
+import { InputField } from '@components/InputField'
+import { DropdownField } from '@containers/TransferCurrency/DropdownField'
+import { FilloutField } from '@containers/TransferCurrency/FilloutField'
+import { addMetalColour } from '@helpers/walletUtils'
+import { getCurrentConnection } from '@selectors'
+import { RootState } from '@store'
+import { Currency, NotificationType, RootRoutes } from '@types'
+
+const mapStateToProps = (state: RootState) => {
+  const { connections, transfer, contacts } = state
+
+  return {
+    ...transfer.formData,
+    ...transfer.formMeta,
+    currency: state.connections.currentCurrency,
```

Comment:
it would be nice to add `accounts` to the destructured values from above, and then use these values for these `state.` calls  eg:

```typescript
  currency: connections.currentCurrency
  balance: accounts.accountInfo.balance
```
etc

Reply:
we could even do the destructuring inside the parameter definition and save the extra line doing the same thing eg,

```typescript
  const mapStateToProps = ({ accounts, connections, contracts, transfer }: RootState) => {
```

Reply:
refactored it